src/information_panel.rs

```diff
@@ -0,0 +1,459 @@
+use egui::Ui;
+use indexmap::{IndexMap, IndexSet};
+use std::fs::File;
+use std::io::{self, Read};
+use std::path::PathBuf;
+
+type SupportedPreviewFilesMap = HashMap<String, Box<dyn FnMut(&mut Ui, &InfoPanelEntry)>>;
+type SupportedPreviewImagesMap =
+    HashMap<String, Box<dyn FnMut(&mut Ui, &InfoPanelEntry, &mut IndexSet<String>)>>;
+type SupportedAdditionalMetaFilesMap =
+    HashMap<String, Box<dyn FnMut(&mut IndexMap<String, String>, &PathBuf)>>;
+
+fn format_pixels(pixels: u32) -> String {
+    const K: u32 = 1_000;
+    const M: u32 = K * 1_000;
+
+    if pixels >= K {
+        format!("{:.2} MPx", f64::from(pixels) / f64::from(M))
+    } else {
+        format!("{pixels} Px")
+    }
+}
+
+/// Wrapper for the `DirectoryEntry` struct, that also adds the option to store text content
+#[derive(Debug)]
+pub struct InfoPanelEntry {
+    /// Directory Item containing info like path
+    pub directory_entry: DirectoryEntry,
+    /// Optional text content of the file
+    pub content: Option<String>,
+}
+
+impl InfoPanelEntry {
+    /// Create a new `InfoPanelEntry` object
+    pub const fn new(item: DirectoryEntry) -> Self {
+        Self {
+            directory_entry: item,
+            content: None,
+        }
+    }
+}
+
+impl InfoPanelEntry {
+    /// Returns the content of the directory item, if available
+    pub fn content(&self) -> Option<&str> {
+        self.content.as_deref()
+    }
+
+    /// Mutably borrow content
+    pub fn content_mut(&mut self) -> &mut Option<String> {
+        &mut self.content
+    }
+}
+
+/// The `InformationPanel` struct provides a panel to display metadata and previews of files.
+/// It supports text-based file previews, image previews, and displays file metadata.
+///
+/// # Fields
+/// - `load_text_content`: Flag to control whether text content should be loaded for preview.
+/// - `supported_files`: A hashmap mapping file extensions to their respective preview rendering functions.
+pub struct InformationPanel {
+    panel_entry: Option<InfoPanelEntry>,
+    /// Flag to control whether text content should be loaded for preview.
+    pub load_text_content: bool,
+    /// Max chars that should be loaded for preview of text files.
+    pub text_content_max_chars: usize,
+    /// Path of the current item that is selected
+    loaded_file_name: PathBuf,
+    /// Map that contains the handler for specific file types (by file extension)
+    supported_preview_files: SupportedPreviewFilesMap,
+    /// Map that contains the handler for image types (by file extension)
+    supported_preview_images: SupportedPreviewImagesMap,
+    /// Map that contains the additional metadata loader for specific file types (by file extension)
+    additional_meta_files: SupportedAdditionalMetaFilesMap,
+    /// Other metadata (loaded by the loader in `additional_meta_files`)
+    other_meta_data: IndexMap<String, String>,
+    /// Stores the images already loaded by the egui loaders.
+    stored_images: IndexSet<String>,
```

Comment:
Is there anything wrong with making it a `Vec`? Should improve performance

Reply:
Could result in double-entries when a file gets selected again. We would have to check each time if the entry is already in the `Vec`, and I think a `HashSet` provides better performance than that, right?

Reply:
I think with 10 items in the Vec the performance will be the same if we iterate and compare strings.

But ye, then the `IndexMap` can stay.